### ISSUE TITLE
support for CohereForAI/c4ai-command-r-v01

### ISSUE THREAD

**Issue by xusenlinzy** [vllm-project/vllm#3330]
[CohereForAI/c4ai-command-r-v01](https://huggingface.co/CohereForAI/c4ai-command-r-v01) is a large language model with open weights optimized for a variety of use cases including reasoning, summarization, and question answering. Command-R has the capability for multilingual generation evaluated in 10 languages and highly performant RAG capabilities.

**Comment by simon-mo** [comment]
Contribution welcomed! https://docs.vllm.ai/en/latest/models/adding_model.html

**Comment by sungkim11** [comment]
I have read it on Twitter that Cohere will support this effort. Are they involved?

**Comment by dekelcohen** [comment]
:+1:
Is there some one who is already working on this ? 
I saw recent activity in transformers repo to add Cohere Command-R.

**Comment by dekelcohen** [comment]
Command-R arch is very close to Lllama2. almost identical code in HF
LayerNorm is different compared to Lllama RMSNorm

**Comment by Playerrrrr** [comment]
So does vLLM support it now or not?

**Comment by simon-mo** [comment]
It's supported in main branch, which you can use when building from source. It will be in next release (ETA tomorrow)

**Comment by 0x77dev** [comment]
Looks like it was released in [v0.4.0](https://github.com/vllm-project/vllm/releases/tag/v0.4.0)

**Comment by mohamad-tohidi** [comment]
how can we use the special RAG capabilities of this model?

for example


```python
# Define conversation input
conversation = [{"role": "user", "content": "What has Man always dreamed of?"}]

# Define documents for retrieval-based generation
documents = [ 
  {"heading": "The Moon: Our Age-Old Foe", "body": "Man has always dreamed of destroying the moon. In this essay, I shall..."},
  {"heading": "Love is all you need", "body": "Man's dream has always been to find love. This profound lesson..."},
]

# Get the RAG prompt
input_prompt = tokenizer.apply_chat_template(
  conversation=conversation,
  documents=documents,
  tokenize=False,
  add_generation_prompt=True,
  return_tensors="pt",
)

# Tokenize the prompt
input_ids = tokenizer.encode_plus(input_prompt, return_tensors="pt")

```

this is from their model card
how is that for vLLM openai api?
